bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,447 @@
+            } catch (IOException e) {
+                logger.warn("Problem sending command to mylink: {} Message: {}", command, e.getMessage());
+                throw new SomfyMyLinkException("Problem sending command to mylink", e);
+            } catch (InterruptedException e) {
+                logger.debug("Interrupted while waiting after sending command to mylink: {} Message: {}", command,
+                        e.getMessage());
+            }
+        }
+    }
+
+    private @Nullable SomfyMyLinkResponseBase sendCommandWithResponse(String command, Type responseType)
+            throws SomfyMyLinkException {
+        synchronized (CONNECTION_LOCK) {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection();
+                        OutputStream out = socket.getOutputStream();
+                        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
+                    byte[] sendBuffer = command.getBytes(StandardCharsets.US_ASCII);
+
+                    // send the command
+                    out.write(sendBuffer, 0, sendBuffer.length);
+
+                    // now read the reply
+                    char[] readBuff = new char[1024];
+                    int readCount;
+                    String message = "";
+
+                    // while we are getting data ...
+                    while (((readCount = in.read(readBuff)) != -1)) {
+                        logger.debug("Got response. Len: " + readCount);
+                        message += new String(readBuff, 0, readCount);
+
+                        try {
+                            logger.debug("Got message: " + message);
+
+                            JsonParser parser = new JsonParser();
+                            JsonObject jsonObj = parser.parse(message).getAsJsonObject();
+
+                            if (jsonObj != null && jsonObj.has("error")) {
+                                SomfyMyLinkErrorResponse errorResponse = gson.fromJson(message,
+                                        SomfyMyLinkErrorResponse.class);
+
+                                logger.info("Error communicating with mylink: {}", errorResponse.error.message);
+                                throw new SomfyMyLinkException(
+                                        "Error communicating with mylink: " + errorResponse.error.message);
+                            }
+
+                            return gson.fromJson(message, responseType);
+                        } catch (JsonSyntaxException e) {
+                            // it wasn't a full message?
+                            logger.debug("Trouble parsing message received. Message:" + e.getMessage());
+                        }
+                    }
+                }
+
+                // only if we didn't already get a response give time for mylink to process
+                Thread.sleep(CONNECTION_DELAY);
```

Comment:
Is this just to throttle messages sent?

Reply:
correct. it slows down both comms if it was sent a message with no reply. it gives it time to process the message.